src/chapter3/02-writing-the-api.md

```diff
@@ -0,0 +1,320 @@
+```
+
+Note that there are currently no routes present - we will add them in a second.
+
+## An intermezzo regarding UUIDs
+
+The task IDs we use here are so called UUIDs - short for **universally unique identifiers**.
+Such an ID can be easily generated randomly and is, for all practical purposes, unique.
+This means that if you generate a UUID for your application, it has never been generated before - therefore UUIDs are a good way of, well, uniquely identifying a resource.
+There are multiple variants of UUIDs - here we use UUIDv4, which is just a special way of creating UUIDs.
+
+When we create a new task later on, we will need to automatically generate new UUIDs.
+For that we can use the `uuid` package.
+Install it by running:
+
+```javascript
+npm install uuid
+```
+
+Now we can generate UUIDs as follows:
+
+```javascript
+const { v4: uuidv4 } = require('uuid');
+const generatedId = uuidv4();
+console.log(generatedId);
+```
+
+This will print a UUID.
+In fact this is how the IDs of the two tasks in the `tasks` variable were generated.
+
+There are many other versions of unique IDs, some of which we will return to later.
+
+## REST and CRUD
+
+There are two words you will hear very often when people talk about web APIs and the like - REST and CRUD.
+
+**REST** is short for "Representational State Transfer" and outlines guidelines for managing communications between clients and servers.
+If you read about REST online, there will be a lot of verbose language like "layered system architecture" or "code on demand".
+However, at its core a REST API is about writing requests that ask for resources using some kind of uniform resource identifier (like UUIDs) and receiving a representation of those resources.
+For example, a client might ask for the task with the UUID `d7b74e8f-34bc-4c11-9260-9369102c53f6` and receive a representation of the respective task which might look as follows:
+
+```json
+{
+  "summary": "Write a task app",
+  "description": "Write an awesome task app."
+}
+```
+
+**CRUD** stands for "Create, Read, Update, Delete" and describes the four basic operations most simple web APIs will support.
+For example, we might want to:
+
+- create a new task
+- read the contents of a task (by its UUID)
+- update a task (e.g. with a new summary or description)
+- delete a task
+
+CRUD maps very nicely to HTTP, because you can use
+
+- POST requests for creating
+- GET requests for reading
+- PUT requests for updating (although POST is commonly used as well for this purpose)
+- DELETE requests for deleting
+
+Together REST and CRUD will guide the design of our web API.
+
+## GET routes to obtain the tasks
+
+We will write two GET routes.
+The `/task` GET route will return all tasks as a JSON dictionary.
+The `/task/:id` GET route will return the task by its ID.
+For example `/task/d7b74e8f-34bc-4c11-9260-9369102c53f6` will return a JSON representing the first task.
+Here is the `/task` route:
+
+```javascript
+app.get('/task', (req, res) => {
```

Comment:
it may be appropriate to name name the args `request` and `response`

Reply:
`req` and `res` and convention in Node.js world, see e.g. the Express docs